modules/core/src/main/java/org/apache/synapse/api/AbstractApiHandler.java

```diff
@@ -51,19 +51,30 @@ protected boolean dispatchToAPI(Collection<API> apiSet, MessageContext synCtx) {
 
         Object apiObject = synCtx.getProperty(RESTConstants.PROCESSED_API);
         if (apiObject != null) {
-            if (identifyAPI((API) apiObject, synCtx, defaultStrategyApiSet)) {
+            if (startProcess((API) apiObject, synCtx, defaultStrategyApiSet)) {
                 return true;
             }
         } else {
             //To avoid apiSet being modified concurrently
             List<API> duplicateApiSet = new ArrayList<API>(apiSet);
-            for (API api : duplicateApiSet) {
-                if (identifyAPI(api, synCtx, defaultStrategyApiSet)) {
+            API identifiedApi = (API) synCtx.getProperty(RESTConstants.IDENTIFIED_API);
+            //If normal path this goes to else, if logging enabled then goes inside the if method.
```

Comment:
This logic is not clear to me, can you please explain?

Reply:
In the api logging flow in the first iteration it needs to identify the api. For that it calls the `identifyApi()` method. If the api is identified then that api gets stored in the IDENTIFIED_API property. Then in the second iteration, we do not need to identify the api again. Therefore we get the identified api from the from the IDENTIFIED_API property. So we do not need to iterate all the apis in the duplicateApiSet to identify the api.

But if the api logging is disabled then api is not identified previously, therefore IDENTIFIED_API property is null, So the api needs to be identified by iterating all the apis in the defaultStrategyApiSet.